zuulcilint/__main__.py

```diff
@@ -93,70 +132,113 @@ def get_all_zuul_yaml_files(files: list[str]) -> list[pathlib.Path]:
     return zuul_yaml_files
 
 
-def get_all_jobs(zuul_yaml_files: list[pathlib.Path]) -> list[list[str]]:
-    """Get all jobs from Zuul YAML files."""
-    all_jobs = []
+def get_all_zuul_objects_by_type(
+    zuul_yaml_files: List[pathlib.Path],
+    zuul_obj: ZuulObject,
+) -> List[Union[dict, None]]:
+    """Get all Zuul objects from provided Zuul YAML files.
+
+    Args:
+    ----
+        zuul_yaml_files: A list of Zuul YAML files.
+        zuul_obj: A ZuulObject enum.
+
+    Returns:
+    -------
+        A list of Zuul objects.
+    """
+    all_zuul_objects = []
     for file_path in zuul_yaml_files:
-        jobs = zuul_utils.get_zuul_object_from_yaml(ZuulObject.JOB, file_path)
-        all_jobs.append([job.get(ZuulObject.JOB.value, {}).get("name") for job in jobs])
-    return all_jobs
+        zuul_objects = zuul_utils.get_zuul_object_from_yaml(zuul_obj, file_path)
+        all_zuul_objects.extend(zuul_objects)
+    return all_zuul_objects
 
 
 def print_warnings(
-    bad_yml_files: list[str],
-    duplicated_jobs: set[str],
+    warnings: dict,
     severity: MsgSeverity = MsgSeverity.WARNING,
 ) -> None:
-    """Print warnings."""
-    n_bad = len(bad_yml_files)
-    n_dup = len(duplicated_jobs)
+    """Print warnings.
+
+    Args:
+    ----
+        warnings: A dictionary containing warnings.
+        severity: A MsgSeverity enum.
+
+    Returns:
+    -------
+        None.
+    """
+    n_bad_yaml = len(warnings["warnings"]["bad_yaml_files"])
+    n_duplicate = len(warnings["warnings"]["duplicated_jobs"])
+    n_nodeset = len(warnings["warnings"]["inexistent_nodesets"])
 
-    if n_bad == 0 and n_dup == 0:
+    if n_bad_yaml == 0 and n_duplicate == 0 and n_nodeset == 0:
         return
 
     if severity == MsgSeverity.WARNING:
         zuul_utils.print_bold("Warnings", MsgSeverity.WARNING)
-        print(f"Total {severity.value}s: {n_bad + n_dup}")
+        print(f"Total {severity.value}s: {n_duplicate + n_duplicate + n_nodeset}")
 
-    if bad_yml_files:
+    if n_bad_yaml:
         zuul_utils.print_bold(f"File extension {severity.value}s:", severity)
         zuul_utils.print_bold(
-            f"Found {n_bad} files with 'yml' extension",
+            f"Found {n_bad_yaml} files with 'yml' extension",
             None,
         )
-        for file_path in bad_yml_files:
+        for file_path in warnings["warnings"]["bad_yaml_files"]:
             print(f"{file_path}")
-    if duplicated_jobs:
+    if n_duplicate:
```

Comment:
Add a newline before this if statement for readability.

Reply:
newline added